src/dart/lib/widgets.dart

```diff
@@ -0,0 +1,288 @@
+// Copyright 2021 The IconVG Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:ui' show Picture, Rect;
+
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'decoder.dart';
+
+/// Render the specified IconVG [asset] as an icon.
+///
+/// This honors the ambient [IconTheme].
+///
+/// The [IconThemeData.color] is used as a one-color custom palette.
+class IconVG extends StatelessWidget {
+  const IconVG(this.asset, {
+    Key? key,
+  }) : super(key: key);
+
+  /// The name of the asset to render.
+  ///
+  /// The ambient [DefaultAssetBundle] is used to resolve this name.
+  final String asset;
+
+  @override
+  Widget build(BuildContext context) {
+    final IconThemeData theme = IconTheme.of(context);
+    return Opacity(
+      opacity: theme.opacity!,
```

Comment:
Is there any way we can pass this through the palette? I'm assuming no because we might have to blend things in some complicated manner (e.g. draw paths of multiple alpha values over each other and then alpha blend the whole thing at the end). But this will be expensive :\

Reply:
Yeah.